### DISCUSSION TITLE
Implementing `Dynamic Speed-UP Mode`?

### DISCUSSION THREAD

**Discussion by bobqianic** [ggml-org/whisper.cpp#1208] [Ideas]
Often, I've noticed that even when a speaker is talking slowly, background music can disrupt things. This tends to make our current speed-up feature diminish the quality of the voice transcription. What if we could detect sections where the background music is causing interference and keep the transcription at normal speed, while speeding up the rest? Moreover, it would be great if we could gauge the speaker's pace and avoid speeding up when they're already speaking quickly. What's your take on this? Any thoughts or suggestions?

**Comment by ggerganov** [comment]
I feel like such type of rule-based algorithms are bound to fail in the general case. The beauty of Whisper is that it does not have and conditional statements. When we start adding `if`s and `else`s, it will become messy.

In any case, examples of such approaches can be added for demonstration purposes. The only limitation is that they will likely not become part of the core `whisper.cpp` library
